src/main/resources/liquibase/crop_changelog/21_3_0.xml

```diff
@@ -0,0 +1,814 @@
+			</sqlCheck>
+		</preConditions>
+		<comment>Add foreign key 'stockprop_cvterm_cvterm_id_fk' to 'stockprop.cvalue_id'</comment>
+		<sql>
+			ALTER TABLE stockprop
+					ADD CONSTRAINT stockprop_cvterm_cvterm_id_fk FOREIGN KEY (cvalue_id)
+					REFERENCES cvterm (cvterm_id);
+		</sql>
+	</changeSet>
+
+	<!-- ENTRY_CODE related changesets -->
+	<changeSet author="lkovacic" id="v21.3.0-27">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT count(1) FROM cvtermprop WHERE cvterm_id = 8300 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+			</sqlCheck>
+		</preConditions>
+		<comment>Change 'ENTRY_CODE' variable type to entry detail</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE cvtermprop SET value = 'Entry Detail' WHERE cvterm_id = 8300 AND type_id = 1800 AND value = 'Germplasm Descriptor';
+
+			INSERT INTO entry_detail_migration_output (cvterm_id) VALUES (8300);
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-28">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT count(1) > 0 FROM projectprop WHERE variable_id = 8300 AND type_id = 1804;
+			</sqlCheck>
+		</preConditions>
+		<comment>Change 'ENTRY_CODE' variable type to entry detail in 'projectprop' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE projectprop SET type_id = 1815 WHERE variable_id = 8300 AND type_id = 1804;
+		</sql>
+	</changeSet>
+
+	<changeSet author="cuenyad" id="v21.3.0-29">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="0">
+				SELECT count(1) > 0 FROM  list_data_details WHERE variable_id = 8300;
+			</sqlCheck>
+		</preConditions>
+		<comment>Migrate ENTRY_CODE from listdata to list_data_details table as entry details variable</comment>
+		<sql dbms="mysql" splitStatements="true">
+			<!-- Remove ENTRY_CODE as default view -->
+			DELETE FROM list_data_default_view WHERE name = 'ENTRY_CODE';
+
+			<!-- Remove ENTRY_CODE as static column -->
+			DELETE FROM list_data_view WHERE static_id = 8300;
+
+			<!-- Migrate ENTRY_CODE from listdata to list_data_details -->
+			INSERT INTO list_data_details(variable_id, lrecid, value, created_by)
+			SELECT 8300, r.lrecid, r.entrycd, ln.listuid FROM listdata r inner join listnms ln on ln.listid = r.listid;
+
+			<!-- Insert ENTRY_CODE variable to list_data_view -->
+			INSERT INTO list_data_view(listid,cvterm_id,type_id)
+			SELECT distinct(ld.listid), 8300, 1815 FROM listnms ld;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="v21.3.0-30">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+				SELECT COUNT(1) > 0
+				FROM project p
+						 INNER JOIN projectprop pp ON p.project_id = pp.project_id AND p.dataset_type_id = 4
+						 INNER JOIN stock s ON p.parent_project_id = s.project_id
+						 INNER JOIN stockprop sp ON s.stock_id = sp.stock_id
+				WHERE pp.variable_id = 8300 AND sp.type_id = 8300;
```

Comment:
@lik-83 seems that type_id = 1815 is missing in both precondition and query

Reply:
Done